### PR TITLE
Setup for separate Linux and OSX packages/services

### PR DESCRIPTION
**Pull request by iAmMrinal0**
Had to rethink the folder and file structure as I could only use `home-manager` on the MacBook due to access restrictions so this structure is a way towards using all user programs in a simpler way between Linux and OSX.

### CODE REVIEW

File: system/common.nix [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  extras = (lib.optionalAttrs (builtins.currentSystem == systems.darwin) {
+    nixpkgs = osx.nixpkgs;
+  }) // (lib.optionalAttrs (builtins.currentSystem == systems.linux) {
+    gtk = linux.gtk;
+    xsession = linux.xsession;
+    qt = linux.qt;
+    services = linux.services;
+    systemd = linux.systemd;
+  });
+}
```
Comment (06kellyjac) [inline system/common.nix:107]: There are helpers in `stdenv` called `isDarwin` and `isLinux` to reduce `builtins.currentSystem == systems.darwin` etc.
Also you've already got `lib` available, no need to pull it out of `pkgs` again

```
  inherit (pkgs.stdenv) isDarwin isLinux;
in {
  programs = lib.recursiveUpdate (lib.recursiveUpdate programs
    (lib.optionalAttrs (isLinux)
      linux.programs))
    (lib.optionalAttrs (isDarwin) osx.programs);
  home = {
    packages = home.packages
      ++ (lib.optionals (isLinux)
        linux.home.packages)
      ++ (lib.optionals (isDarwin)
        osx.home.packages);
    sessionVariables = lib.recursiveUpdate home.sessionVariables
      (lib.optionalAttrs (isDarwin)
        osx.home.sessionVariables);
  };
  extras = (lib.optionalAttrs (isDarwin) {
    nixpkgs = osx.nixpkgs;
  }) // (lib.optionalAttrs (isLinux) {
    gtk = linux.gtk;
    xsession = linux.xsession;
    qt = linux.qt;
    services = linux.services;
    systemd = linux.systemd;
  });
}
```

File: system/linux.nix [hunk 2]
```diff
@@ -0,0 +1,139 @@
+{ lib, pkgs, ... }:
+
+let
+  keepmenu = pkgs.callPackage ../pkgs/keepmenu { };
+  rofimoji = pkgs.callPackage ../pkgs/rofimoji { };
+
+  wallpaper = lib.readFile (pkgs.callPackage ../scripts/wallpaper.nix { });
+
+  i3blocksConf = pkgs.callPackage ../config/i3blocks.nix { };
+  packages = [
```
Comment (06kellyjac) [inline system/linux.nix:10]: `packages = with pkgs; [` then you can remove `pkgs.` from the front of everything within the array

File: system/common.nix [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+let
+  systems = {
+    darwin = "x86_64-darwin";
+    linux = "x86_64-linux";
+  };
+  linux = import ./linux.nix { inherit lib pkgs; };
+  osx = import ./osx.nix { inherit pkgs; };
+  zshCustom = pkgs.callPackage ../config/modSteeefZsh.nix { };
+
+  packages = [
```
Comment (06kellyjac) [inline system/common.nix:12]: `packages = with pkgs; [` then you can remove `pkgs.` from the front of everything within the array

File: system/common.nix (hunk 1 above)
Comment (iAmMrinal0) [reply system/common.nix:107]: yeah, I am trying to use the `isDarwin` and `isLinux` from `stdenv` but I ended up getting infinite recursion errors. there is some discussion going on in matrix #nix:nixos.org around this.
and thanks for catching the `pkgs.lib` and `lib`. will change it.

File: system/linux.nix (hunk 2 above)
Comment (iAmMrinal0) [reply system/linux.nix:10]: thanks, makes sense.

**Pull request merged**